src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -51,15 +59,68 @@ const Zmenu = (props: ZmenuProps) => {
 
   const destroyZmenu = () => {
     dispatch(changeHighlightedTrackId(''));
-    setZmenus && setZmenus(pickBy(zmenus, (value, key) => key !== props.id));
+    setZmenus &&
+      setZmenus(pickBy(zmenus, (value, key) => key !== props.zmenuId));
   };
 
   const direction = chooseDirection(props);
   const toolboxPosition =
     direction === Direction.LEFT ? ToolboxPosition.LEFT : ToolboxPosition.RIGHT;
 
+  const { content, variety } = props.payload;
```

Comment:
I am not sure I understand what is happening below. I am also not sure I understand the distribution of responsibilities between components anymore.

We now find ourselves with the following arrangement:
1. Genome browser sends a zmenu message
2. There is some code that listens to the message, creates a new zmenu per message, keeps track of the created zmenus, and removes them when they are no longer needed. Previously this code lived in ZmenuController; but by now it is shared between ZmenuController and useGenomeBrowser in a way that I don't quite understand
3. Next, a newly instantiated zmenu needs to decide what kind of zmenu it is. We expect that for transcript, variant, and so on, zmenus are going to be different; this is what the `variety` field suggests. So, by inspecting the variety key, we should be able to select the appropriate zmenu component to display the content of the message. I would expect us to have a `ZmenuTranscript` component for handling the `gene-and-one-transcript` zmenu type.

Here are the questions for us:
- Does the `ZmenuController` component have any purpose, or can its logic be moved into a hook (e.g. `useZmenus`). By the way, does the zmenu-related code belong in the useGenomeBrowser hook at all; or is it better placed in such useZmenus hook?
-  What's the responsibility of the `Zmenu` component? Should it only inspect the `variety` property and choose the appropriate zmenu component for handling this variety, or is it also responsible for the zmenu positioning logic? Or should the positioning logic be extracted into a hook?